app/src/behaviors/behavior_sensor_rotate.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2022 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_sensor_rotate
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+
+#include <drivers/sensor.h>
+#include <zmk/event_manager.h>
+#include <zmk/events/keycode_state_changed.h>
+#include <zmk/keymap.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT)
```

Comment:
There are some recent improvements to Zephyr (documented in https://docs.zephyrproject.org/2.6.0/guides/kconfig/preprocessor-functions.html) that provide a more efficient way to do this.

First add to app/Kconfig something like
```
DT_COMPAT_ZMK_BEHAVIOR_SENSOR_ROTATE := zmk,behavior-sensor-rotate

config ZMK_BEHAVIOR_SENSOR_ROTATE
    bool
    default $(dt_compat_enabled,$(DT_COMPAT_ZMK_BEHAVIOR_SENSOR_ROTATE))
```

and then in CMakeLists.txt you can use
```cmake
target_sources_ifdef(CONFIG_ZMK_BEHAVIOR_SENSOR_ROTATE app PRIVATE src/behaviors/behavior_sensor_rotate.c)
```
so that this whole file just gets excluded from the build if the behavior isn't used anywhere. Then you don't need to conditionally compile the contents of it.